### PR TITLE
[breaking] Reading shares and roles for grants

### PR DESCRIPTION
**Pull request by alldoami**
Grants (_grant) resources must now be unique for their corresponding entity. We are doing this since the `Read` functions on grants were not working properly (for example, when terraform importing resources).

### CODE REVIEW

File: pkg/resources/grant_helpers.go [hunk 1]
```diff
@@ -152,45 +147,29 @@ func readGenericGrant(data *schema.ResourceData, meta interface{}, builder snowf
...
-			}
 			roles = append(roles, grant.GranteeName)
 		case "SHARE":
-			// Shares get the account appended to their name, remove this
-			granteeNameStrippedAccount := StripAccountFromName(grant.GranteeName)
-			if !stringInSlice(granteeNameStrippedAccount, sharesIn) {
-				continue
-			}
-
 			shares = append(shares, grant.GranteeName)
```
Comment (edulop91) [inline pkg/resources/grant_helpers.go]:
```suggestion
			shares = append(shares, granteeNameStrippedAccount)
```
I think that was one of the fixes from before. Maybe keep that line and the comment. but just remove the `if !stringInSlice.....`

File: main.go [hunk 2]
```diff
@@ -40,6 +41,9 @@ func doc() {
 	for _, name := range names {
 		resource := resources[name]
 		fmt.Printf("\n### %s\n\n", name)
+		if strings.Contains(name, "_grant") {
+			fmt.Printf("%s are unique for a given %s\n\n", name, strings.Trim(name, "_grant"))
```
Comment (edulop91) [inline main.go]:
```suggestion
			fmt.Printf(`\`\`\`
			%s are unique for a given %s
			\`\`\`
			`, name, strings.Trim(name, "_grant"))
```

File: README.md [hunk 3]
```diff
@@ -81,6 +81,7 @@ You can see a number of examples [here](examples).
 
 ### snowflake_database_grant
 
+**Note**: snowflake_database_grant are unique for a given snowflake_database.
```
Comment (ryanking) [inline README.md]: I think we need to expand on this a bit.

File: main.go [hunk 4]
```diff
@@ -40,6 +41,16 @@ func doc() {
 	for _, name := range names {
 		resource := resources[name]
 		fmt.Printf("\n### %s\n\n", name)
+		if strings.HasSuffix(name, "_grant") {
+			grant_resource_name := strings.Trim(name, "_grant")
+			fmt.Printf(`
+			**Note**: The grant resource create exclusive attachments of grants.
```
Comment (edulop91) [inline main.go]:
```suggestion
			**Note**: The grant resource creates exclusive attachments of grants.
```

File: main.go [hunk 5]
```diff
@@ -40,6 +41,16 @@ func doc() {
...
+			Across the entire AWS account, all of the users/roles/groups to which a single grant is attached must be declared
```
Comment (edulop91) [inline main.go]: `users/roles/groups` might not make sense in this context -> I wonder if you can fetch it from the schema that is available in the context

File: main.go [hunk 6]
```diff
@@ -40,6 +41,16 @@ func doc() {
...
+			by a single %s resource. This means that even any %s that have the attached
+			grant via any other mechanism (including other Terraform resources) will have that attached grant revoked by this resource.
+			These resources do not enforce exclusive attachment of a grant.\n
```
Comment (edulop91) [inline main.go]:
```suggestion
			These resources do not enforce exclusive attachment of a grant, it is the user's responsibility to enforce this.\n
```

**Pull request merged**